python/ray/tests/test_metrics.py

```diff
@@ -377,6 +379,444 @@ def getpid(self):
     assert profiling_stats is not None
 
 
+def test_memory_dashboard(shutdown_only):
+    """Test Memory table.
+
+    These tests are motivated by this document.
+    https://docs.ray.io/en/latest/memory-management.html#debugging-using-ray-memory
+    """
+    addresses = ray.init(num_cpus=2)
+    webui_url = addresses["webui_url"].replace("localhost", "http://127.0.0.1")
+    assert (wait_until_server_available(addresses["webui_url"]) is True)
+
+    def get_memory_table():
+        memory_table = requests.get(webui_url + "/api/memory_table").json()
+        return memory_table["result"]
+
+    def memory_table_ready():
+        memory_table = get_memory_table()
+        return len(memory_table["group"]) > 0
+
+    def stop_memory_table():
+        requests.get(webui_url + "/api/stop_memory_table").json()
+        time.sleep(1.0)
+
+    def test_local_reference():
+        @ray.remote
+        def f(arg):
+            return arg
+
+        # a and b are local references.
+        a = ray.put(None)
+        b = f.remote(None)
+
+        wait_for_condition(memory_table_ready)
+        memory_table = get_memory_table()
+        summary = memory_table["summary"]
+        group = memory_table["group"]
+        assert summary["total_captured_in_objects"] == 0
+        assert summary["total_pinned_in_memory"] == 0
+        assert summary["total_used_by_pending_task"] == 0
+        assert summary["total_local_ref_count"] == 2
+        for table in group.values():
+            for entry in table["entries"]:
+                assert (
+                    entry["reference_type"] == ReferenceType.LOCAL_REFERENCE)
+        stop_memory_table()
+        del a
+        del b
```

Comment:
`a` and `b` are going to be deleted when this function returns. We don't do these explicitly here. Same as `del a,b` invocation belows